### PR TITLE
add description about limitation

### PR DESCRIPTION
**Pull request by D3Hunter**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [x] v8.3 (TiDB 8.3 versions)
- [x] v8.2 (TiDB 8.2 versions)
- [x] v8.1 (TiDB 8.1 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/18774
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-change-column.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ ERROR 8200 (HY000): Unsupported modify column: change from original type decimal
 * Changes of [Reorg-Data](/sql-statements/sql-statement-modify-column.md#reorg-data-change) types on primary key columns are not supported.
 * Changes of column types on partitioned tables are not supported.
 * Changes of column types on generated columns are not supported.
-* Changes of some data types (for example, some TIME, Bit, Set, Enum, and JSON types) are not supported due to the compatibility issues of the `CAST` function's behavior between TiDB and MySQL.
+* Changes from some data types (for example, some TIME, Bit, Set, Enum, and JSON types) to some other types are not supported due to the compatibility issues of the `CAST` function's behavior between TiDB and MySQL.
```
Comment (dveeden) [inline sql-statements/sql-statement-change-column.md:155]:
```suggestion
* Changes from some data types (for example, TIME, BIT, SET, ENUM, and JSON types) to some other types are not supported due to the compatibility issues of the `CAST` function's behavior between TiDB and MySQL.
```

The text is not very clear if this is only the `TIME` data type or all time related data types.

Maybe link to `data-type-<type>.md` ? 

Maybe link to `functions-and-operators/cast-functions-and-operators.md` ?

File: tidb-limitations.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ aliases: ['/docs/dev/tidb-limitations/']
 
 This document describes the common usage limitations of TiDB, including the maximum identifier length and the maximum number of supported databases, tables, indexes, partitioned tables, and sequences.
 
+Note: TiDB is highly compatible with the MySQL protocol, and also compatible with many limitations of MySQL, such as a maximum of 16 columns is permitted for multicolumn indexes, for more information, see [MySQL Compatibility](/mysql-compatibility.md) and offical doc of MySQL.
```
Comment (dveeden) [inline tidb-limitations.md:11]:
```suggestion
Note: TiDB is highly compatible with the MySQL protocol and syntax, and also compatible with many limitations of MySQL, such as a maximum of 16 columns is permitted for multicolumn indexes, for more information, see [MySQL Compatibility](/mysql-compatibility.md) and official documentation of MySQL.
```

File: sql-statements/sql-statement-change-column.md (hunk 1 above)
Comment (D3Hunter) [reply sql-statements/sql-statement-change-column.md:155]: yes, but we don't have a clear description doc about what's the compatibility issues is. we need someone familiar with that part to write a doc about it.

json/enum column do can modify to be varchar, so we add this one, previously we are saying no modify on those type can be done,

**Review by hfxsd — APPROVED** [review]

File: sql-statements/sql-statement-modify-column.md [hunk 3]
```diff
@@ -205,7 +205,7 @@ CREATE TABLE `t1` (
     ERROR 8200 (HY000): Unsupported modify column: table is partition table
     ```
 
-* Does not support modifying some data types (for example, some TIME types, Bit, Set, Enum, JSON) are not supported due to some compatibility issues of the `cast` function's behavior between TiDB and MySQL.
+* Does not support modifying from some data types (for example, some TIME types, Bit, Set, Enum, JSON) to some other types due to some compatibility issues of the `cast` function's behavior between TiDB and MySQL.
```
Comment (hfxsd) [inline sql-statements/sql-statement-modify-column.md:208]:
```suggestion
* Does not support modifying from some data types (for example, some TIME types, BIT, SET, ENUM, JSON) to some other types due to some compatibility issues of the `cast` function's behavior between TiDB and MySQL.
```

File: tidb-limitations.md [hunk 4]
```diff
@@ -8,6 +8,10 @@ aliases: ['/docs/dev/tidb-limitations/']
 
 This document describes the common usage limitations of TiDB, including the maximum identifier length and the maximum number of supported databases, tables, indexes, partitioned tables, and sequences.
 
+> **Note:**
+>
+> TiDB offers high compatibility with the MySQL protocol and syntax, including many MySQL limitations. For example, a multicolumn index in TiDB also supports a maximum of 16 columns. For more information, see [MySQL Compatibility](/mysql-compatibility.md) and the official MySQL documentation.
```
Comment (hfxsd) [inline tidb-limitations.md:13]:
```suggestion
> TiDB offers high compatibility with the MySQL protocol and syntax, including many MySQL limitations. For example, For example, a single index can include a maximum of 16 columns. For more information, see [MySQL Compatibility](/mysql-compatibility.md) and the official MySQL documentation.
```

**Pull request merged**